### DISCUSSION TITLE
Service to expose entities to Assist / Google Assistant

### DISCUSSION THREAD

**Discussion by TheFes** [frenck/spook#370] [Ideas]
With Christmas coming up, it would be nice if it was possible to have a service call to expose entities to Assist /Google Assistant.
So when I enable `input_boolean.christmas` I can immediately turn on the Christmas tree using voice.

I do realize this is already possible by hiding the Christmas related entities. But in other use cases it might be nice to keep the entity visible and not expose them sometimes.

**Comment by rwelsh09** [comment]
I too would like to be able to automatically hide/expose entities from/to voice assistants. 🤞

**Comment by itsgreat2misha** [comment]
I'd also greatly benefit from this-- there doesn't appear to be a way to programmatically expose entities to assist today, which complicates things in complex environments like mine where many of my entities are created automatically and it would be a major pain to go through each one and expose manually one at a time.
